app/src/main/java/org/simple/clinic/registration/facility/RegistrationFacilitySelectionUiRenderer.kt

```diff
@@ -8,12 +10,17 @@ import org.simple.clinic.location.LocationUpdate.Unavailable
 import org.simple.clinic.mobius.ViewRenderer
 import org.simple.clinic.registration.RegistrationConfig
 
-class RegistrationFacilitySelectionUiRenderer(
-    private val ui: RegistrationFacilitySelectionUi,
+class RegistrationFacilitySelectionUiRenderer @AssistedInject constructor(
+    @Assisted private val ui: RegistrationFacilitySelectionUi,
     private val facilityListItemBuilder: FacilityListItemBuilder,
     private val registrationConfig: RegistrationConfig
 ) : ViewRenderer<RegistrationFacilitySelectionModel> {
 
+  @AssistedInject.Factory
```

Comment:
Is it a new approach we will be implementing for UiRenderer?

Reply:
We will only use this in places where the UI renderer needs access to dependencies that are injectable.